core/src/log.cpp

```diff
@@ -0,0 +1,29 @@
+#include <knoting/log.h>
+
+#include <spdlog/sinks/stdout_color_sinks.h>
+
+namespace knot {
+
+namespace log {
+
+void Logger::setup() {
+#ifdef KNOTING_DEBUG
+    log::set_level(spdlog::level::debug);
```

Comment:
seems reasonable to only display messages in debug
but what happens when a user issues log::debug message in release?

Reply:
They just get filtered out, so they are not printed to any sink we have included.